docs/proposals/lifelong-learning.md

```diff
@@ -0,0 +1,185 @@
+    
+## Controller Design
+
+The Lifelong learning controller starts three separate goroutines called `upstream`, `downstream` and `Lifelonglearningjob`controller.<br/>
+These are not separate controllers as such but named here for clarity.
+- Lifelong learning: watch the updates of lifelong-learning job crds, and create the workers depending on the state machine.
+- downstream: synchronize the lifelong-learning-job updates from the cloud to the edge node.
+- upstream: synchronize the lifelong-learning-job updates from the edge to the cloud node.
+
+### Lifelong Learning Controller
+![](./images/lifelong-learning-controller.png)
+
+The lifelong-learning controller watches for the updates of lifelong-learning jobs and the corresponding pods against the K8S API server.<br/>
+Updates are categorized below along with the possible actions:
+
+| Update Type                    | Action                                       |
+|-------------------------------|---------------------------------------------- |
+|New lifelong-learning-job Created             | Wait to train trigger satisfied|
+|lifelong-learning-job Deleted                 | NA. These workers will be deleted by [k8s gc](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).|
+|The Status of lifelong-learning-job Updated               | Create the train/eval worker if it's ready.|
+|The corresponding pod created/running/completed/failed                 | Update the status of lifelong-learning job.|
+
+### Downstream Controller
+![](./images/lifelong-learning-downstream-controller.png)
+
+The downstream controller watches for the lifelong-learning job updates against the K8S API server.<br/>
+Updates are categorized below along with the possible actions that the downstream controller can take:
+
+| Update Type                    | Action                                       |
+|-------------------------------|---------------------------------------------- |
+|New Lifelong-learning-job Created             |Sends the job information to LCs.|
+|Lifelong-learning-job Deleted                 | The controller sends the delete event to LCs.|
+
+### Upstream Controller
+![](./images/lifelong-learning-upstream-controller.png)
+
+The upstream controller watches for the lifelong-learning job updates from the edge node and applies these updates against the API server in the cloud.<br/>
+Updates are categorized below along with the possible actions that the upstream controller can take:
+
+| Update Type                        | Action                                        |
+|-------------------------------     |---------------------------------------------- |
+|Lifelong-learning-job Reported State Updated    |  The controller appends the reported status of the job by LC in the cloud. |
+
+### Details of api between GM(cloud) and LC(edge)
+1. GM(downstream controller) syncs the job info to LC:
```

Comment:
same with the section of incremental learning job, please use a link instead of copy